source/src/microapp/cs_MicroappController.cpp

```diff
@@ -136,16 +145,18 @@ void jumpToAddress(uintptr_t address) {
  * microapp forever. Alternatively, if the function never yields, it will trip the watchdog. If the watchdog is
  * triggered, we might presume that a microapp was the reason for it and disable it.
  *
- * @param[in] void *p                            pointer to a coroutine_args_t struct
+ * @param[in] void *args    Arguments provided when starting the coroutine.
  */
-void goIntoMicroapp(void* p) {
+void goIntoMicroapp(void* args) {
 
 #if DEVELOPER_OPTION_USE_DUMMY_CALLBACK_IN_BLUENET == 1
 	LOGi("Call main: %p", &microappCallbackDummy);
 	microappCallbackDummy();
 #else
-	coroutine_args_t* coargs = (coroutine_args_t*)p;
-	LOGi("Within coroutine, call main: %p", coargs->entry);
+	// TODO: this is microapp context, but this reads data from bluenet stack (args == &_sharedState).
```

Comment:
Yes, but `coargs` is at a special place at the end of RAM that is accessible by both bluenet and microapp.

Reply:
coargs is, but coargs->entry is not.

Reply:
But `coargs->entry` should be just a field.

Actually we should just jump in the cs_DoubleCoroutine.c file instead of calling this function in the microapp controller. Then we don't need this argument passed around either.